src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -171,40 +199,57 @@ def backfill_seer_grouping_records(
                 )
             )
         if response["success"]:
+            groups_with_neighbor = response["groups_with_neighbor"]
             groups = Group.objects.filter(project_id=project.id, id__in=group_id_batch)
             for group in groups:
+                seer_similarity = {
+                    "similarity_model_version": SEER_SIMILARITY_MODEL_VERSION,
+                    "request_hash": group_hashes_dict[group.id],
+                }
+                if str(group.id) in groups_with_neighbor:
+                    # TODO: remove this try catch once the helper is made
+                    try:
+                        seer_similarity["results"] = [
+                            asdict(
+                                SeerSimilarIssueData.from_raw(
+                                    project_id, groups_with_neighbor[str(group.id)]
+                                )
+                            )
+                        ]
+                    except (IncompleteSeerDataError, SimilarGroupNotFoundError) as e:
+                        logger.exception(
+                            str(e),
+                            extra={
+                                "project_id": project_id,
+                                "group_id": group.id,
+                                "hash": group_hashes_dict[group.id],
+                            },
+                        )
+
                 if group.data.get("metadata"):
-                    group.data["metadata"]["embeddings_info"] = {
-                        "nn_model_version": 0,
-                        "group_hash": json.dumps([group_hashes_dict[group.id]]),
-                    }
+                    group.data["metadata"]["seer_similarity"] = seer_similarity
```

Comment:
if there is an exception in the above, won't this fail as seer_similarity won't be defined?

Reply:
it won't fail since it's been defined [here](https://github.com/getsentry/sentry/pull/70992/files#diff-073ef410c8c0c0aa12ee71c34018b70edd325d37bf684273293b3db96397b293R216-R218), but it won't be updated to show that a record wasn't created for it. this logic is a bit broken until we start deleting records. I think the correct logic here would be to not update the metadata at all so we can try to backfill again later once we support record deletion